src/vte.c

```diff
@@ -675,6 +675,45 @@ const gchar *vte_get_working_directory(void)
 				g_free(cwd);
 			}
 		}
+		else
+		{
+			gint status = 0;
+			gchar *stdout = NULL;
+			gchar *pid_str = g_strdup_printf("%d", pid);
+			gchar *argv[] = {"lsof", "-a", "-d", "cwd", "-F", "n", "-p", pid_str, NULL};
+
+			if (utils_spawn_sync(NULL, argv, NULL, G_SPAWN_SEARCH_PATH,
+					NULL, NULL, &stdout, NULL, &status, NULL))
+			{
+				if (status == 0)
+				{
+					gboolean pid_verified = FALSE;
+					gchar **line;
+					gchar **lines = g_strsplit(stdout, "\n", -1);
+
+					/* stdout is of the form: pPID\nnPATH (PID and PATH are actual values) */
+					foreach_strv(line, lines)
+					{
+						g_strstrip(*line);
+						if (strlen(*line) > 1 && *line[0] == 'p')
```

Comment:
the `strlen()`s here are basically useless, if `line[0] == 'p'` the next will at worse be `\0`, which the code in the block handles just fine